src/main/scala/akka/rabbitmq/RabbitMqActor.scala

```diff
@@ -18,13 +17,8 @@ trait RabbitMqActor extends Actor with ShutdownListener {
     self ! AmqpShutdownSignal(cause)
   }
 
-  type ChannelOrConnection = {
-    def isOpen(): Boolean
-    def close(): Unit
-  }
-
-  def closeIfOpen(x: ChannelOrConnection) {
-    if (x.isOpen()) x.close()
+  def close(x: AutoCloseable): Unit = try x.close() catch {
```

Comment:
@reinierl We were not catching exceptions before. That could be a reason for channel/connection actors to restart.

Reply:
But yeah, while it's definitely sloppy coding in the original, it still doesn't provide a good explanation of the resource leak. If calling `close` throws an exception, then well, we can't close it, which cuts off our way from properly disposing of the resource. We're fried either way in that case.